sdk/core/src/request_options/ba512_range.rs

```diff
@@ -102,27 +104,27 @@ mod test {
     }
 
     #[test]
-    #[should_panic(expected = "ParseIntError(ParseIntError { kind: InvalidDigit })")]
     fn test_512range_parse_panic_1() {
-        "abba/2000".parse::<BA512Range>().unwrap();
+        let err = "abba/2000".parse::<BA512Range>().unwrap_err();
+        assert_eq!(err.to_string(), "parse int error: invalid digit found in string".to_string());
```

Comment:
Is there a reason we're matching against the formatted string representation of the error instead of the error type itself?

Reply:
@rylev Hi, Thanks reviewing!
At first I also would try that matching these error types according with your mentions
```
--- a/sdk/core/src/request_options/range.rs
+++ b/sdk/core/src/request_options/range.rs
@@ -109,6 +109,7 @@ impl<'a> AddAsHeader for Range {
 #[cfg(test)]
 mod test {
     use super::*;
+    use std::num::IntErrorKind::InvalidDigit;

     #[test]
     fn test_range_parse() {
@@ -120,8 +121,9 @@ mod test {

     #[test]
     fn test_range_parse_panic_1() {
-        let err = "abba/2000".parse::<Range>().unwrap_err();
-        assert_eq!(err.to_string(), "Parse int error invalid digit found in string".to_string());
+        let err = "abba/2000".parse::<Range>();
+        let pie = ParseIntError{kind : InvalidDigit};
+        assert_eq!(err, Err(ParseError::ParseIntError(pie)));
     }
```
But, It became a compile failed for this reasons. I'm not sure that explicitly...
```
% cargo test
   Compiling azure_core v0.1.0 (/Users/yugo-horie/azure-sdk-for-rust/sdk/core)
error[E0658]: use of unstable library feature 'int_error_matching': it can be useful to match errors when making error messages for integer parsing
   --> sdk/core/src/request_options/range.rs:112:9
    |
112 |     use std::num::IntErrorKind::InvalidDigit;
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #22639 <https://github.com/rust-lang/rust/issues/22639> for more information

error[E0658]: use of unstable library feature 'int_error_matching': it can be useful to match errors when making error messages for integer parsing
   --> sdk/core/src/request_options/range.rs:125:40
    |
125 |         let pie = ParseIntError{kind : InvalidDigit};
    |                                        ^^^^^^^^^^^^
    |
    = note: see issue #22639 <https://github.com/rust-lang/rust/issues/22639> for more information

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0658`.
error: could not compile `azure_core`
```

Reply:
You can do:

```rust
assert!(matches!(err, Err(ParseError::ParseIntError(_)))
```

This means you don't have to construct the actual `ParseIntError` type.